### PR TITLE
add axes support for dropouts in gluon

### PR DESCRIPTION
**Pull request by szha**
## Description ##
add axes support for dropouts in gluon

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] add dropout axes support in Dropout, VariationalDropoutCell

### CODE REVIEW

File: python/mxnet/gluon/nn/basic_layers.py [hunk 1]
```diff
@@ -239,15 +241,16 @@ class Dropout(HybridBlock):
         `Dropout: A Simple Way to Prevent Neural Networks from Overfitting
         <http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf>`_
     """
-    def __init__(self, rate, **kwargs):
+    def __init__(self, rate, axes=(), **kwargs):
```
Comment (sxjscience) [inline python/mxnet/gluon/nn/basic_layers.py]: Default to None?

Comment (szha) [reply python/mxnet/gluon/nn/basic_layers.py]: This is the same default value as the one added in Dropout op. I set it this way so that I don't have to handle None.

File: tests/python/unittest/test_gluon.py [hunk 2]
```diff
@@ -827,6 +827,46 @@ def selu(x):
...
+                ix += (slice(None, None, None),)
+        return x[ix]
+
+    def check_dropout_axes(ratio, shape, axes):
+        compactshape = list(shape)
+        for axis in axes:
+            compactshape[axis] = 1
+        compactx = mx.random.uniform(shape=tuple(compactshape))
+        broadcastx = compactx.broadcast_to(shape)
+        dropouty = mx.gluon.nn.Dropout(rate=ratio, axes=axes)(broadcastx)
```
Comment (sxjscience) [inline tests/python/unittest/test_gluon.py]: May need consider to improve the test in the future. Currently there's no guarantee that the observed dropout ratio matches the given dropout ratio.
